### PR TITLE
Milestone5

### PR DESCRIPTION
**Pull request by vidhishweta01**
Added clickHandler to app, button panel, button
logic is now connected to UI of the calculator

### CODE REVIEW

File: src/component/App/App.js [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
+  }
+
+  handleClick = (buttonName) => {
+    this.setState((prevState) => calculate(prevState, buttonName));
+  };
 
+  render() {
+    const { total, next } = this.state;
+    return (
+      <>
```
Comment (george-shammar) [inline src/component/App/App.js:23]: - [ ] Please, always remember to make your commits in the present imperative tenses. You have committed your codes in past tense. 
- [ ] Please, do not disable eslint as you did here on line 12. There's a simple work around for this eslint error. Please, let's fix it. :pray: 
- [ ] I got this error `Error: [big.js] Division by zero` when i tried to divide a number by zero. Please, let's handle this exception.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @vidhishweta01 , 

Good job so far!

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 
@vidhishweta01 Nice job done on the suggested changes, I can't find the requested changes but however, few changes are required to get your project approved
### highlights
Git flow 👍🏻 
### Improve 

- [x] button `%` still not working. kindly fix 
- [x] your division by 0 should display `Division by Zero error` not  ` something went wrong` so that the user can know that it's an error.

### once specific changes are made kindly request for another review
### Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

**Comment by vidhishweta01** [comment]
https://www.loom.com/share/fe3dfde4691942c186f4a50e84d5d1ca

modulus operator is working completely fine please check it properly  and you can watch this video given in above link

File: src/logic/calculate.js [hunk 2]
```diff
@@ -54,45 +64,62 @@ export default function calculate(obj, buttonName) {
...
-  if (buttonName === '=') {
-    if (isNumber(total) && !isNumber(next)) {
+
+  // Button must be an operation
+
+  // When the user presses an operation button without having entered
+  // a number first, do nothing.
+  // if (!obj.next && !obj.total) {
+  //   return {};
+  // }
```
Comment (rindrajosia) [inline src/logic/calculate.js:97]: - [ ]  [OPTIONAL] Inline comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history.

File: src/logic/operate.js [hunk 3]
```diff
@@ -3,19 +3,20 @@ import Big from 'big.js';
...
     return ONE.plus(TWO).toString();
   } if (operation === '-') {
     return ONE.minus(TWO).toString();
-  } if (operation === '÷') {
+  } if (operation === '÷' && !(TWO > -1 && TWO < 1)) {
     return ONE.div(TWO).toString();
-  } if (operation === 'x') {
-    return ONE.times(TWO).toString;
-  } if (operation === '%') {
-    return ONE.mod(TWO).toString();
```
Comment (rindrajosia) [inline src/logic/operate.js:16]: - [ ]  [OPTIONAL] In fact, this operator should return percentage, not modulo.

**Review by rindrajosia — APPROVED** [review]
Hi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**